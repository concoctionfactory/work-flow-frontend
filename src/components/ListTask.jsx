import React, { useState } from "react";
import { useDispatch } from "react-redux";
import Task from "./Task";
import ListTaskForm from "./ListTaskForm";
import TaskFrom from "./TaskForm";
import { removeListTaskAPI } from "../actions/boards";

import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import IconButton from "@material-ui/core/IconButton";
import { red } from "@material-ui/core/colors";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import AddIcon from "@material-ui/icons/Add";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
const useStyles = makeStyles((theme) => ({
  root: {
    width: 345,
    margin: theme.spacing(1),
  },
  media: {
    height: 0,
    paddingTop: "56.25%", // 16:9
  },
  expand: {
    transform: "rotate(0deg)",
    marginLeft: "auto",
    transition: theme.transitions.create("transform", {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: "rotate(180deg)",
  },
  avatar: {
    backgroundColor: red[500],
  },
  add: {
    marginLeft: "auto",
  },
}));

function ListTask({ list }) {
  const { id, name, cards, board_id } = list;
  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const dispatch = useDispatch();

  function handleDelete() {
    dispatch(removeListTaskAPI(list));
    handleClose();
  }

  const [taskForm, setTaskForm] = useState({
    isOpen: false,
  });

  const openTaskFrom = () => {
    handleClose();
    setTaskForm({ isOpen: true });
  };

  const closeTaskFrom = () => {
    setTaskForm({ isOpen: false });
  };

  const [listTaskForm, setListTaskForm] = useState({
    isOpen: false,
  });

  const openListTaskFrom = () => {
    handleClose();
    setListTaskForm({ isOpen: true });
  };

  const closeListTaskFrom = () => {
    setListTaskForm({ isOpen: false });
  };
  return (
    <Card className={classes.root}>
      <CardHeader
        action={
          <IconButton
            aria-label="settings"
            aria-haspopup="true"
            onClick={handleClick}
          >
            <MoreVertIcon />
          </IconButton>
        }
        title={name}
      />

      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem onClick={handleDelete}>Delete</MenuItem>
        <MenuItem onClick={openListTaskFrom}>Edit</MenuItem>
      </Menu>

      <CardContent>
        <Grid container spacing={1}>
          {cards &&
            Object.entries(cards).map(([key, value]) => (
              <Grid item key={key} xs={12}>
                <Task key={key} card={value} boardId={board_id}></Task>
              </Grid>
            ))}
        </Grid>
      </CardContent>
      <CardActions disableSpacing>
        <Button
          aria-label="add"
          color="primary"
          variant="contained"
          onClick={openTaskFrom}
          className={classes.add}
        >
          <AddIcon />
        </Button>
      </CardActions>

      <ListTaskForm
        className={classes.fab}
        isOpen={listTaskForm.isOpen}
        handleOpen={openListTaskFrom}
        handleClose={closeListTaskFrom}
        boardId={board_id}
        listTask={list}
      ></ListTaskForm>

      <TaskFrom
        className={classes.fab}
        isOpen={taskForm.isOpen}
        handleOpen={openTaskFrom}
        handleClose={closeTaskFrom}
        listTaskId={id}
        boardId={board_id}
      ></TaskFrom>
    </Card>
  );
}

export default ListTask;
