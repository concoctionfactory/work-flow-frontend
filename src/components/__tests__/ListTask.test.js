import React from "react";
import { render } from "@testing-library/react";
import ListTask from "../ListTask";
import { userData, boardData } from "./testData";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "../../reducers/root";
import { BrowserRouter } from "react-router-dom";
import thunk from "redux-thunk";
const store = createStore(rootReducer, applyMiddleware(thunk));
const list = boardData.lists[0];

it("renders without crashing", function () {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <ListTask list={list} />
      </BrowserRouter>
    </Provider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <ListTask list={list} />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows text ", function () {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <ListTask list={list} />
      </BrowserRouter>
    </Provider>
  );
  expect(component.getByText("Delete")).toBeInTheDocument();
  expect(component.getByText("Edit")).toBeInTheDocument();
});
