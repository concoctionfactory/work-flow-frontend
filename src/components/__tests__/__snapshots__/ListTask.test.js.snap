// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-9 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          design
        </span>
      </div>
      <div
        class="MuiCardHeader-action"
      >
        <button
          aria-haspopup="true"
          aria-label="settings"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-16 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-gutterBottom"
                >
                  create wireframe
                </span>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                >
                  use figma instead of sketch
                </h6>
              </div>
              <div
                class="MuiCardHeader-action"
              >
                <button
                  aria-label="edit"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing"
            >
              <div
                class="MuiChip-root MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  not_started
                </span>
              </div>
              <div
                class="MuiChip-root MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  Oct 1, 2020
                </span>
              </div>
            </div>
            <div />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-16 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-gutterBottom"
                >
                  mockup mobile
                </span>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                >
                  pass off using zeplin
                </h6>
              </div>
              <div
                class="MuiCardHeader-action"
              >
                <button
                  aria-label="edit"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing"
            >
              <div
                class="MuiChip-root MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  in_progress
                </span>
              </div>
              <div
                class="MuiChip-root MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  Oct 6, 2020
                </span>
              </div>
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiCardActions-root"
    >
      <button
        aria-label="add"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-add-14 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div />
    <div />
  </div>
</DocumentFragment>
`;
